scipy/stats/_multicomp.py

```diff
@@ -0,0 +1,263 @@
+           8.32
+
+    >>> import numpy as np
+    >>> control = np.array([7.40, 8.50, 7.20, 8.24, 9.84, 8.32])
+    >>> drug_a = np.array([9.76, 8.80, 7.68, 9.36])
+    >>> drug_b = np.array([12.80, 9.68, 12.16, 9.20, 10.55])
+
+    The `dunnett` statistic is sensitive to the difference in means between
+    the samples.
+
+    We would like to see if the means between any of the groups are
+    significantly different. First, visually examine a box and whisker plot.
+
+    >>> import matplotlib.pyplot as plt
+    >>> fig, ax = plt.subplots(1, 1)
+    >>> ax.boxplot([control, drug_a, drug_b])
+    >>> ax.set_xticklabels(["Control", "Drug A", "Drug B"])  # doctest: +SKIP
+    >>> ax.set_ylabel("mean")  # doctest: +SKIP
+    >>> plt.show()
+
+    From the box and whisker plot, we can see overlap in the interquartile
+    ranges between the control group and the group from drug A.
+    We can apply the `dunnett`
+    test to determine if the difference between means is significant. We
+    set a significance level of .05 to reject the null hypothesis.
+
+    >>> from scipy.stats import dunnett
+    >>> res = dunnett(drug_a, drug_b, control=control)
+    >>> res.pvalue
+    array([0.47773146, 0.00889328])  # random
+
+    The null hypothesis is that each group has the same mean. The p-value for
+    comparisons between ``control`` and ``drug_b`` do not exceed .05,
+    so we reject the null hypothesis that they
+    have the same means. The p-value of the comparison between ``control``
+    and ``drug_a`` exceeds .05, so we accept the null hypothesis that there
+    is not a significant difference between their means.
+
+    """
+    samples_, control, rng = iv_dunnett(
+        samples=samples, control=control, random_state=random_state
+    )
+
+    rho, df, n_group = params_dunnett(samples=samples_, control=control)
+
+    statistic = np.array([
+        stats.ttest_ind(
+            obs_, control, alternative=alternative, random_state=rng
+        ).statistic
+        for obs_ in samples_
+    ])
+
+    pvalue = pvalue_dunnett(
+        rho=rho, df=df,
+        statistic=statistic, alternative=alternative,
+        rng=rng
+    )
+
+    return DunnettResult(
+        statistic=statistic, pvalue=pvalue,
+    )
+
+
+def iv_dunnett(
+    samples: Sequence[npt.ArrayLike],
+    control: npt.ArrayLike,
+    random_state: SeedType
+) -> tuple[list[np.ndarray], np.ndarray, SeedType]:
+    """Input validation for Dunnett's test."""
+    rng = check_random_state(random_state)
+
+    ndim_msg = "Control and samples groups must be 1D arrays"
+    n_obs_msg = "Control and samples groups must have at least 1 observation"
+
+    control = np.asarray(control)
+    samples_ = [np.asarray(sample) for sample in samples]
+
+    # samples checks
+    samples_control: list[np.ndarray] = samples_ + [control]
+    for sample in samples_control:
+        if sample.ndim > 1:
+            raise ValueError(ndim_msg)
+
+        if len(sample) < 1:
```

Comment:
I think we'll get an unexpected error for scalars. Simple fix, since it's definitely a NumPy array:
```suggestion
        if sample.size < 1:
```